adapters/repos/db/shard_backup.go

```diff
@@ -80,7 +80,20 @@ func (s *Shard) HaltForTransfer(ctx context.Context, offloading bool, inactivity
 		q.Pause()
 		return nil
 	})
+	// wait for ongoing indexing to finish
+	_ = s.ForEachVectorQueue(func(_ string, q *VectorIndexQueue) error {
+		q.Wait()
```

Comment:
line above we are pausing the indexing with `q.Pause()` isn't that enough to then just do `q.Flush()`? or we do have to actually `q.Wait()` before `q.Flush()`?